src/ds/cli_helper.h

```diff
@@ -19,11 +19,45 @@ namespace cli
     std::string port = {};
   };
 
+  bool parse_address(
+    const std::string_view& addr,
+    ParsedAddress& parsed,
+    const std::string& option_name,
+    const std::string& default_port = "0")
+  {
+    auto found = addr.find_last_of(":");
+    auto hostname = addr.substr(0, found);
+
+    const auto port =
+      found == std::string::npos ? default_port : addr.substr(found + 1);
+
+    // Check if port is in valid range
+    int port_int;
+    try
```

Comment:
uint16_t port_int = 0;
std::from_chars(port_str.data(), port_str.data() + port_str.size(), port_int);
...

to save the manual validation perhaps?